### PR TITLE
Add 'More' option to the table column context menu

### LINKED ISSUE #9432: Refine right click menu
Follow-up to https://github.com/JabRef/jabref/issues/6601

We take Windows Explorer as reference (see https://github.com/JabRef/jabref/issues/6601#issuecomment-724932377)

Current appearance:

grafik

Only columns enabled in the preferences are shown:

grafik

How to enable addition of new columns (as shown at the screenshot at https://github.com/JabRef/jabref/issues/6601#issuecomment-724932377)?

We think "More..." ("Mehr..." in German) is missing:

grafik

After a click, a new popup opens 

grafik

Would be nice if that was added here, too. Would that be possible?

This list should be filled with all fields known to JabRef. In case a user wants to add more fields, they must go to the preferences and add the custom field there.

### PR DESCRIPTION
**Pull request by nikolaynikolaevn**
Fixes #9432 

This PR adds a "More" option to the context menu that appears when a table column is right-clicked. The idea is to ease the addition of new columns. When the "More" option is selected, the Preferences window opens at the "Entry table" tab, so that the existing functionality for managing the table columns shown can be reused.

image
image

```[tasklist]
### Compulsory checks
- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.
```

### CODE REVIEW

File: src/main/java/org/jabref/gui/maintable/MainTableHeaderContextMenu.java [hunk 1]
```diff
@@ -224,4 +234,20 @@ public boolean isVisibleInTable() {
...
+     *
+     */
+    private class RightClickMenuMoreItem extends MenuItem {
+
+        RightClickMenuMoreItem(String displayName) {
+            super(displayName);
+
+            // Set action to open the Preferences > Entry table window
+            this.setOnAction(event -> {
+                new ShowPreferencesAction(JabRefGUI.getMainFrame(), Globals.TASK_EXECUTOR, new TableTab()).execute();
```
Comment (calixtus) [inline src/main/java/org/jabref/gui/maintable/MainTableHeaderContextMenu.java:249]: Please do not use JabRefGui, Globals and JabRefFrame. Aim for loose coupling.

Comment (koppor) [reply src/main/java/org/jabref/gui/maintable/MainTableHeaderContextMenu.java:249]: Would it be a good thing to provide these parameters at the constructor - or would you rather go for a lambda?

File: src/main/java/org/jabref/gui/maintable/MainTableHeaderContextMenu.java [hunk 2]
```diff
@@ -75,6 +80,11 @@ private void constructItems(MainTable mainTable) {
             RightClickMenuItem itemToAdd = createMenuItem(tableColumn, false);
             this.getItems().add(itemToAdd);
         }
+
+        this.getItems().add(separator);
+
+        RightClickMenuMoreItem itemToAdd = new RightClickMenuMoreItem("More");
```
Comment (koppor) [inline src/main/java/org/jabref/gui/maintable/MainTableHeaderContextMenu.java:86]: UI pattern: Use three dots if another menu optens:

```suggestion
        RightClickMenuMoreItem itemToAdd = new RightClickMenuMoreItem("More...");
```

**Review by koppor — CHANGES_REQUESTED** [review]
From the look-and-feel, it sounds good. We need to work on "good" code though.

File: src/main/java/org/jabref/gui/maintable/MainTableHeaderContextMenu.java (hunk 1 above)
Comment (nikolaynikolaevn) [reply src/main/java/org/jabref/gui/maintable/MainTableHeaderContextMenu.java:250]: Unfortunately, I am not sure how to resolve this.

File: src/main/java/org/jabref/gui/maintable/MainTableHeaderContextMenu.java [hunk 3]
```diff
@@ -75,6 +80,12 @@ private void constructItems(MainTable mainTable) {
             RightClickMenuItem itemToAdd = createMenuItem(tableColumn, false);
             this.getItems().add(itemToAdd);
         }
+
+        separator = new SeparatorMenuItem();
+        this.getItems().add(separator);
+
+        RightClickMenuMoreItem itemToAdd = new RightClickMenuMoreItem("More...");
```
Comment (koppor) [inline src/main/java/org/jabref/gui/maintable/MainTableHeaderContextMenu.java:87]: We need `Localization.lang("More...")` here, don't we?

File: src/main/java/org/jabref/gui/maintable/MainTableHeaderContextMenu.java (hunk 1 above)
Comment (koppor) [reply src/main/java/org/jabref/gui/maintable/MainTableHeaderContextMenu.java:250]: image

File: src/main/java/org/jabref/gui/preferences/ShowPreferencesAction.java [hunk 4]
```diff
@@ -11,15 +11,32 @@ public class ShowPreferencesAction extends SimpleCommand {
...
 
     public ShowPreferencesAction(JabRefFrame jabRefFrame, TaskExecutor taskExecutor) {
         this.jabRefFrame = jabRefFrame;
         this.taskExecutor = taskExecutor;
     }
 
+    /**
+     * Creates an action to open a preferences window at a specific tab
+     * @param jabRefFrame
+     * @param taskExecutor
```
Comment (koppor) [inline src/main/java/org/jabref/gui/preferences/ShowPreferencesAction.java:24]: Remove params without explanation

**Review by koppor — CHANGES_REQUESTED** [review]
I tried to guess  how @calixtus meant "loose coupling"

**Comment by Siedlerchr** [comment]
Closing this due to inactivity

**Pull request closed without merging**